### PR TITLE
chore: Table refactor

### PR DESCRIPTION
**Pull request by richburdon**
- organize components into sub folders (move next to react-ui-table)
- factor out common types and utils with grid (e.g., GridCell)
- simplify handlers and seed constructors
- TODOs

### CODE REVIEW

File: packages/plugins/plugin-table/src/model/table-model.ts [hunk 1]
```diff
@@ -86,7 +88,7 @@ export class TableModel extends Resource {
       return Object.fromEntries(
         fields.map((field, index: number) => {
           return [
-            getCellKey(index, 0),
+            fromCellKey({ col: index, row: 0 }),
```
Comment (ZaymonFC) [inline packages/plugins/plugin-table/src/model/table-model.ts:91]: I find this name confusing, it doesn't indicate what's being created.

File: packages/sdk/schema/src/types/types.ts [hunk 2]
```diff
@@ -83,7 +83,7 @@ export const QuerySchema = S.Struct({
 
 // TODO(burdon): Are views always flat projections?
 export const ViewSchema = S.Struct({
-  query: QuerySchema,
+  query: S.optional(QuerySchema),
```
Comment (ZaymonFC) [inline packages/sdk/schema/src/types/types.ts:86]: Nice, I've been wondering about this.

**Review by ZaymonFC — APPROVED** [review]
Looks good to me.

File: packages/plugins/plugin-table/src/model/table-model.ts (hunk 1 above)
Comment (richburdon) [reply packages/plugins/plugin-table/src/model/table-model.ts:91]: ok to rename this but follow existing standards (see sheet) and make consistent (i.e., toX, fromX, etc.)

Comment (ZaymonFC) [reply packages/plugins/plugin-table/src/model/table-model.ts:91]: Works for me. Agree get was suboptimal.

Comment (ZaymonFC) [reply packages/plugins/plugin-table/src/model/table-model.ts:91]: A pattern I have used in the past is defining to and from as object properties. Then you can invoke them like:

`gridCellKey.fromCoordinates({ row, col })` or something like that.

**Pull request merged**